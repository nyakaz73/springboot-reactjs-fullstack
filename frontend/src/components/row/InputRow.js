import React, { Children } from 'react';
import PropTypes from "prop-types";
const InputRow = ({ children, label }) => {
    const singleChild = Children.only(children);
    const hasInputElement = Children.map(children, child => {
        if (child.type === 'input')
            return true;
        return false;
    });
    return (
        <div style={container}>
            <p style={labelStyle}>{label}</p>
            <div className="styledInput" >
                {
                    hasInputElement[0] ? singleChild : new Error('an input element is required!')
                }
            </div>
        </div>
    );
}

const container = {
    display: 'flex',
    flexDirection: 'column',
    padding: '10px'
}
const labelStyle = {
    color: '#1010108a',
    fontWeight: '500',
    paddingLeft: '16px'
}

InputRow.propTypes = {
    children: PropTypes.node.isRequired,
    label: PropTypes.string.isRequired
}
export default InputRow;
