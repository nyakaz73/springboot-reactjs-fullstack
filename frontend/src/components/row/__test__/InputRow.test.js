import React from "react";
import ReactDOM from "react-dom";
import { act } from "@testing-library/react";
import InputRow from "../InputRow";
import renderer from "react-test-renderer"

let container;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    document.body.removeChild(container);
    container = null;
});

it("Renders without crashing", () => {
    act(() => {
        ReactDOM.render(<InputRow />, container);
    });
})