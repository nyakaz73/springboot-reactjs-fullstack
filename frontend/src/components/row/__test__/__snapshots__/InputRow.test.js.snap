// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "padding": "10px",
    }
  }
>
  <p
    style={
      Object {
        "color": "#1010108a",
        "fontWeight": "500",
        "paddingLeft": "16px",
      }
    }
  >
    Hello
  </p>
  <div
    className="styledInput"
  >
    <input />
  </div>
</div>
`;
